app/assets/javascripts/graph.js

```diff
@@ -0,0 +1,98 @@
+var StateId = document.getElementById("circle_pack").getAttribute('data-state-id');
+
+$.ajax({
+         type: "GET",
+         contentType: "application/json; charset=utf-8",
+         url: '/states/' + StateId,
+         dataType: 'json',
+         success: function (data) {
+             draw(data);
+         },
+         error: function (result) {
+             error();
+         }
+ });
+
+function draw(data) {
+  var $circle_pack = $('#circle_pack');
+  var StateId = $circle_pack.data('state-id');
+
+  var margin = 20,
+    diameter = 960;
+
+var color = d3.scale.linear()
+    .domain([-1, 5])
+    .range(["hsl(152,80%,80%)", "hsl(228,30%,40%)"])
+    .interpolate(d3.interpolateHcl);
+
+var pack = d3.layout.pack()
+    .padding(2)
+    .size([diameter - margin, diameter - margin])
+    .value(function(d) { return d.size; })
+
+var svg = d3.select("body").append("svg")
+    .attr("width", diameter)
+    .attr("height", diameter)
+  .append("g")
+    .attr("transform", "translate(" + diameter / 2 + "," + diameter / 2 + ")");
+
+d3.json("/states/" + StateId + ".json", function(error, root) {
+  if (error) return console.error(error);
+
+  var focus = root,
+      nodes = pack.nodes(root),
+      view;
+
+  var circle = svg.selectAll("circle")
+      .data(nodes)
+    .enter().append("circle")
+      .attr("class", function(d) { return d.parent ? d.children ? "node" : "node node--leaf" : "node node--root"; })
+      .style("fill", function(d) { return d.children ? color(d.depth) : null; })
+      .on("click", function(d) { if (focus !== d) zoom(d), d3.event.stopPropagation(); });
```

Comment:
Line is too long.<br>Expected '{' and instead saw 'zoom'.<br>Expected an assignment or function call and instead saw an expression.<br>'d3' is not defined.